### ISSUE TITLE
Compilation error when mocking trait with method with bounded type parameter

### ISSUE THREAD

**Issue by DrPhil** [paulbutcher/ScalaMock#254]
### ScalaMock Version
4.1.0

### Scala Version
2.12.8

### Runtime
Scalac - JVM

### Please describe the expected behavior of the issue
`org.scalamock.scalatest.MockFactory` should be able to mock the trait.

### Please provide a description of what actually happens
Compiler error:
```
type mismatch;
 found   : r.type (with underlying type X)
 required: X
Parameter type in structural refinement may not refer to an abstract type defined outside that refinement
Parameter type in structural refinement may not refer to a type member of that refinement
```

### Reproducible Test Case
Scastie: https://scastie.scala-lang.org/lbY7ItfvQa60owC3nhZvfg

```scala
import org.scalamock.scalatest.MockFactory

class SmallSpec extends MockFactory {
  trait Broken[T] {
    def bad[X <: T](r: X): Unit
  }

  val debug = stub[Broken[Unit]]
}
```

**Comment by barkhorn** [comment]
Thanks for raising this. There are a few related problems open at the moment, particularly related to methods with type parameters

**Comment by informarte** [comment]
Using ScalaMock 4.1.0 and Scala 2.12.8, I tried to compile the following code:

```scala
trait Bar[T] {
     def polymorphic[U <: T](x: List[U]): String
}

val barMock = mock[Bar[Int]]

```
Compilation, however, fails:

```
Error:(136, 27) type mismatch;
found   : List[U(in method polymorphic)]
required: List[(some other)U(in method polymorphic)]
```

**Comment by kostaskougios** [comment]
Have a similar issue with
    class MyService[A <: Resource[A]](alphas:Seq[A])
    val service=mock[MyService[Campaign]]


type mismatch;
 found   : Seq[A]
 required: Seq[com.expedia.assets.publisher.tables.Campaign]
  val service=mock[MyService[Campaign]]

**Comment by barkhorn** [comment]
Closed as a duplicate of #76
